Src/CSharpier/ConfigurationFileOptions.cs

```diff
@@ -17,58 +17,68 @@ public class ConfigurationFileOptions
         public bool UseTabs { get; init; }
         public EndOfLine EndOfLine { get; init; }
 
-        public static ConfigurationFileOptions Create(string rootPath, IFileSystem fileSystem)
-        {
-            ConfigurationFileOptions ReadJson(string path)
-            {
-                return JsonConvert.DeserializeObject<ConfigurationFileOptions>(
-                    fileSystem.File.ReadAllText(path)
-                );
-            }
-
-            ConfigurationFileOptions ReadYaml(string path)
+        public static ConfigurationFileOptions Create(
+            string baseDirectoryPath,
+            IFileSystem fileSystem
+        ) {
+            while (true)
```

Comment:
So, instead of doing `baseDirectoryPath = directoryInfo.Parent.FullName` below, you can probably do `directoryInfo = directoryInfo.Parent`. That way, you can have a `while (directoryInfo != null)` instead of `while (true)` which I prefer to avoid whenever possible.